### PR TITLE
Update to crypto secure random number generation

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

File: sdk/internal/uuid/uuid.go [hunk 1]
```diff
@@ -6,45 +6,28 @@
...
 // A UUID representation compliant with specification in RFC 4122 document.
 type UUID [16]byte
 
 // New returns a new uuid using RFC 4122 algorithm.
 func New() UUID {
 	u := UUID{}
-	// Set all bits to randomly (or pseudo-randomly) chosen values.
-	// math/rand.Read() is no-fail so we omit any error checking.
+	// Set all bits to pseudo-random values.
 	// NOTE: this takes a process-wide lock
```
Comment (richardpark-msft) [inline sdk/internal/uuid/uuid.go:26]: Do we know if the 'process-wide lock' bit is still true?

File: sdk/internal/uuid/uuid.go [hunk 2]
```diff
@@ -6,45 +6,28 @@
...
 )
 
-func init() {
-	rand.Seed(time.Now().Unix())
-}
-
 // A UUID representation compliant with specification in RFC 4122 document.
 type UUID [16]byte
 
 // New returns a new uuid using RFC 4122 algorithm.
```
Comment (richardpark-msft) [inline sdk/internal/uuid/uuid.go:22]: I'm not sure about convention but if a function can panic I think it should definitely be documented as that's a very rare occurrence. Maybe it's just taken as a given that though.

CC: @JeffreyRichter , @jhendrixMSFT for idiom check

**Review by richardpark-msft — COMMENTED** [review]
The panic has me 'panic'king a bit because it feels like a big runtime change but I'm not sure how we treat that in the Go SDK.

I think the usage of this code is local enough that we could just inline it into azure-identity (I think this was the package we saw yesterday).

File: sdk/internal/uuid/uuid.go (hunk 1 above)
Comment (seankane-msft) [reply sdk/internal/uuid/uuid.go:26]: According to the [docs](https://pkg.go.dev/math/rand#Read), there is no mention of a process wide lock, I think this note can be removed in this case.

File: sdk/internal/uuid/uuid.go (hunk 2 above)
Comment (seankane-msft) [reply sdk/internal/uuid/uuid.go:22]: [Docs](https://pkg.go.dev/math/rand#Read) mention that this is always a nil error, I think we can ignore the error for now. You will likely need a `//nolint` comment, the linter might complain about not checking a returned error.

Comment (richardpark-msft) [reply sdk/internal/uuid/uuid.go:22]: I think that doc is for math/rand (@RickWinter swapped to [crypto/rand](https://pkg.go.dev/crypto/rand))

Comment (seankane-msft) [reply sdk/internal/uuid/uuid.go:22]: @richardpark-msft thanks for catching that the math/rand docs also make no mention of a long process, but I think we should be returning the error in this case.

Comment (RickWinter) [reply sdk/internal/uuid/uuid.go:22]: I spoke with Jeff on this change and its our conclusion that the error cases should never be hit given how we use the API.  If they are hit then we would want to stop (panic) so users can debug an unexpected and unrecoverable error

**Review by JeffreyRichter — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
Can I get some context on the reasoning for this change?  When we initially created this helper, we explicitly decided against using the crypto RNG as this was for creating UUIDs for things like request IDs (i.e. it was never meant to be a cryptographically secure UUID implementation).  Presumably this invariant was lost along the way and we're now trying to shore this up?  Assuming this is the case, are we sure that the impl as it stands meets that requirement?  Put another way, if we need a cryptographically secure UUID generation facility, should we be rolling our own or pick an existing impl?

**Comment by RickWinter** [comment]
> Can I get some context on the reasoning for this change? When we initially created this helper, we explicitly decided against using the crypto RNG as this was for creating UUIDs for things like request IDs (i.e. it was never meant to be a cryptographically secure UUID implementation). Presumably this invariant was lost along the way and we're now trying to shore this up? Assuming this is the case, are we sure that the impl as it stands meets that requirement? Put another way, if we need a cryptographically secure UUID generation facility, should we be rolling our own or pick an existing impl?

The issue with the current impl is it will result in duplicate request ids if a client sends enough requests.  That results in a less than ideal debugging experience.  The goal of this change is to remove the likelihood we generate duplicate requiest ids

File: sdk/internal/uuid/uuid.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/internal/uuid/uuid.go:26]: It's not in the docs, you have to look at the impl.  For `crypto/rand` the comment might not be applicable.

Comment (jhendrixMSFT) [reply sdk/internal/uuid/uuid.go:26]: For the Unix impl it also takes a lock.

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/15180

File: sdk/internal/uuid/uuid.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/internal/uuid/uuid.go]: This has been fixed to not panic.  I also fixed the `Parse()` function as it was swallowing errors.

**Review by seankane-msft — APPROVED** [review]

**Pull request merged**